capirca/lib/arista_tp.py

```diff
@@ -0,0 +1,1036 @@
+#       0.0.0.0/0                                       << MATCH_INDENT
+#       except 34.64.0.0/26
+#    !
+#    field-set ipv4 prefix dst-hjjqurby6yftqk6fa3xx4fas
+
+# various indentation constants - see above
+INDENT_STR = " " * 3  # 3 spaces
+TERM_INDENT = 2 * INDENT_STR
+MATCH_INDENT = 3 * INDENT_STR
+ACTION_INDENT = 4 * INDENT_STR
+
+DEFAULT_TERM_SUFFIX = "-all-default"
+
+
+# generic error class
+class Error(Exception):
+    pass
+
+
+class TcpEstablishedWithNonTcpError(Error):
+    pass
+
+
+class AristaTpDuplicateTermError(Error):
+    pass
+
+
+class AristaTpFragmentInV6Error(Error):
+    pass
+
+
+class Config(object):
+    """config allows a configuration to be assembled easily.
+
+    when appending to the configuration object, the element should be indented
+    according to the arista traffic-policy style.
+
+    a text representation of the config can be extracted with str().
+
+    attributes:
+    indent: The number of leading spaces on the current line.
+    lines: the text lines of the configuration.
+
+    """
+
+    def __init__(self):
+        self.lines = []
+
+    def __str__(self):
+        return "\n".join(self.lines)
+
+    def Append(self, line_indent, line, verbatim=False):
+        """append one line to the configuration.
+
+        args:
+        - line_indent: config specific spaces prepended to the line
+        - line: the configuratoin string to append to the config.
+        - verbatim: append line without adjusting indentation. Default False.
+        """
+        if verbatim:
+            self.lines.append(line)
+            return
+
+        self.lines.append(line_indent + line.strip())
+
+
+class Term(aclgenerator.Term):
+    """represents an individual AristaTrafficPolicy term.
+
+    useful for the __str__() method. wher literally, everything intersting
+    happens.
+
+    attributes:
+     term: the term object from the policy.
+     term_type: string indicating type of term, inet, inet6 etc.
+     noverbose: boolean to disable verbosity.
+
+    """
+
+    _PLATFORM = "arista_tp"
+    _ACTIONS = {
+        "accept": "",
+        "deny": "drop",
+        "reject": "drop",               # unsupported action, convert to drop
+        "reject-with-tcp-rst": "drop",  # ibid
+        # "next": "continue",
+    }
+    AF_MAP = {
+        "inet": 4,
+        "inet6": 6,
+    }
+    # the following lookup table is used to map between the various types of
+    # filters the generator can render.  as new differences are
+    # encountered, they should be added to this table.  Accessing members
+    # of this table looks like:
+    #  self._TERM_TYPE('inet').get('saddr') -> 'source-address'
+    #
+    # it's critical that the members of each filter type be the same, that is
+    # to say that if _TERM_TYPE.get('inet').get('foo') returns something,
+    # _TERM_TYPE.get('inet6').get('foo') must return the inet6 equivalent.
+    _TERM_TYPE = {
+        "inet": {
+            "addr_fam": "ipv4",
```

Comment:
none of the other terms, other than "addr_fam" probably need to be here, since they all seem to be the same, unless there is a reason to expect that they will change, in which case, please ignore this comment.